Clarify external packages must have source code published (#2172)
home-assistant/developers.home-assistant: docs/api_lib_index.md

One of the foundational rules of Home Assistant is that we do not include any protocol specific code. Instead, this code should be put into a standalone Python library and published to PyPI. This guide will describe how to get started with this!

## Basic library requirements

> - The library must have source distribution packages available, it's not allowd to rely on packages which only have binary distribution packages.

Reviewer: ```suggestion
- The library must have source distribution packages available; it's not allowed to rely on packages that only have binary distribution packages.
```

> - The library must have source distribution packages available<> it's not allowed to rely on packages that only have binary distribution packages.

Reviewer 2: ```suggestion
- The library must have source distribution packages available<> it's not allowed to rely on packages that only have binary distribution packages.
```
Does this mean that the packages also publish their source to PyPI? Didn't know this was a thing

Reviewer 3: Yes, pypi allows publishing both, a wheel contain platform-specific, prebuilt packages (i.e., something that you can just unpack to install) while source distributions requires building it (e.g., using the setup.py) as part of the installation.

You usually publish both source and binary distributions, as wheels avoid the need to build from the source, and can be seen on packages which use native code, for example https://pypi.org/project/numpy/#files . Wheels are usually preferred as you don't need to install the build toolchains just to use the package.

> - Issue trackers must be enabled for external Python libraries that has communication with external devices or services.

Reviewer: ```suggestion
- Issue trackers must be enabled for external Python libraries that communicate with external devices or services.
```

> - If the library is mainly used for Home Assistant and you are a code owner of the integration, it is encouraged to use an issue template picker with links to [Home Assistant Core Issues](https://github.com/home-assistant/core/issues). For example: [zwave-js-server-python - New Issue](https://github.com/home-assistant-libs/zwave-js-server-python/issues/new/choose)

Reviewer 2: ```suggestion
- If the library is mainly used for Home Assistant and you are a code owner of the integration, it is encouraged to use an issue template picker with links to [Home Assistant Core Issues](https://github.com/home-assistant/core/issues). For example: [zwave-js-server-python - New Issue](https://github.com/home-assistant-libs/zwave-js-server-python/issues/new/choose)
- The library and eventual subdependencies are licensed using an [OSI-approved license](https://opensource.org/license). This should be reflected in the metadata of the library.
```

> - If the library is mainly used for Home Assistant and you are a code owner of the integration it is encouraged to use an issue template picker with links to [Home Assistant Core Issues](https://github.com/home-assistant/core/issues). For example: [zwave-js-server-python - New Issue](https://github.com/home-assistant-libs/zwave-js-server-python/issues/new/choose)

Reviewer: ```suggestion
- If the library is mainly used for Home Assistant and you are a code owner of the integration, it is encouraged to use an issue template picker with links to [Home Assistant Core Issues](https://github.com/home-assistant/core/issues). For example: [zwave-js-server-python - New Issue](https://github.com/home-assistant-libs/zwave-js-server-python/issues/new/choose)
```

One of the foundational rules of Home Assistant is that we do not include any protocol specific code. Instead, this code should be put into a standalone Python library and published to PyPI. This guide will describe how to get started with this!

> ## Basic library requirements
> - The library must have source distribution packages available<> it's not allowed to rely on packages that only have binary distribution packages.

Author: ```suggestion
## Basic library requirements

- The library must have source distribution packages available<> it's not allowed to rely on packages that only have binary distribution packages.
```

One of the foundational rules of Home Assistant is that we do not include any protocol specific code. Instead, this code should be put into a standalone Python library and published to PyPI. This guide will describe how to get started with this!

## Basic library requirements

> - The library must have source distribution packages available, it's not allowed to rely on packages that only have binary distribution packages.

Author: ```suggestion
- The library must have source distribution packages available; it's not allowed to rely on packages that only have binary distribution packages.
```

> - The library must have source distribution packages available<> it's not allowed to rely on packages that only have binary distribution packages.

Author: ```suggestion
- The library must have source distribution packages available, it's not allowed to rely on packages that only have binary distribution packages.
```
